modules/src/settings-daemon.py

```
@@ -395,6 +415,104 @@ def delete_key(key):
+
+
+def profile_write(name, assignments, drop=(), must_exist=False):
+    """One read-modify-write on a profile, under the store's own lock.
+
+    `must_exist` belongs to that same critical section, not to an
+    `os.path.exists` in the caller: outside the lock a concurrent
+    /profiles/delete lands between the check and the write, and the write
+    recreates the profile that was just deleted. Returns False when it was
+    gone.
+    """
+    return update(profile_path(name), assignments, profile_header(name),
```

Ruff (0.16.3)

[error] Undefined name `update`

(F821)